components/feature/session/src/main/java/mozilla/components/feature/session/TrackingProtectionUseCases.kt

```diff
@@ -15,51 +15,51 @@ import java.lang.Exception
 /**
  * Contains use cases related to the tracking protection.
  *
- * @param sessionManager the application's [SessionManager].
+ * @param store the application's [BrowserStore].
  * @param engine the application's [Engine].
  */
 class TrackingProtectionUseCases(
-    val sessionManager: SessionManager,
+    val store: BrowserStore,
     val engine: Engine
 ) {
 
     /**
-     * Use case for adding a new [Session] to the exception list.
+     * Use case for adding a new tab to the exception list.
      */
     class AddExceptionUseCase internal constructor(
-        private val sessionManager: SessionManager,
+        private val store: BrowserStore,
         private val engine: Engine
     ) {
         private val logger = Logger("TrackingProtectionUseCases")
 
         /**
-         * Adds a new [session] to the exception list, as a result this session will
-         * not get applied any tracking protection policy.
-         * @param session The [session] that will be added to the exception list.
+         * Adds a new tab to the exception list, as a result this tab will not get applied any
+         * tracking protection policy.
+         * @param tabId The id of the tab that will be added to the exception list.
          */
-        operator fun invoke(session: Session) {
-            val engineSession = sessionManager.getEngineSession(session)
+        operator fun invoke(tabId: String) {
+            val engineSession = store.state.findTabOrCustomTabOrSelectedTab(tabId)?.engineState?.engineSession
```

Comment:
nit: Might make sense to add an extension function in the future .e.g something like `store.findEngineSession(tabId)`, although if we add lots of those shortcuts it will also get messy and confusing.

Reply:
Yeah, I'm curious how this pattern is going to evolve once we need to create `EngineSession`s through `BrowserStore`.